src/main/java/com/danielagapov/spawn/Models/User.java

```diff
@@ -0,0 +1,29 @@
+package com.danielagapov.spawn.Models;
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.io.Serializable;
+import java.util.UUID;
+
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+public class User implements Serializable {
+        @Id
+        @GeneratedValue
+        private UUID id;
+
+        @Column(nullable = false, unique = true) // Ensures the username is unique and not null
```

Comment:
This might be a tricky one. 

Let's say a user signs in through email & pass, google, or apple -> get to "enter your password" screen
At this point, if they close the app, there wouldn't be a way of us remembering that they signed in, so they'd be prompted again, with this constraint. 

But, maybe that's a concern for the mobile side. Let's keep it this way for now, since I don't want to deal with nullable `username` properties anyway; that'd add unnecessary complexity.